Port WebHitTestResultData to the new IPC serialization format

gavin-apple opened the pull request:
#### 39afa29edb2601dcea29f92669101505bd491d0d
<pre>
Port WebHitTestResultData to the new IPC serialization format
<a href="https://bugs.webkit.org/show_bug.cgi?id=248079">https://bugs.webkit.org/show_bug.cgi?id=248079</a>
rdar://102518135

Reviewed by Alex Christensen.

Port WebHitTestResultData to the new IPC serialization format and add
support for SoftLinkedClasses in the GeneratedSerializers.

* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Scripts/generate-serializers.py:
(sanitize_string_for_variable_name):
(decode_type):
(parse_serialized_types):
* Source/WebKit/Shared/WebHitTestResultData.cpp:
(WebKit::WebHitTestResultPlatformData::WebHitTestResultPlatformData):
(WebKit::WebHitTestResultData::WebHitTestResultData):
(WebKit::WebHitTestResultData::~WebHitTestResultData):
(WebKit::WebHitTestResultData::getImageSharedMemoryHandle const):
(WebKit::WebHitTestResultData::encode const): Deleted.
(WebKit::WebHitTestResultData::decode): Deleted.
(WebKit::WebHitTestResultData::platformEncode const): Deleted.
(WebKit::WebHitTestResultData::platformDecode): Deleted.
* Source/WebKit/Shared/WebHitTestResultData.h:
* Source/WebKit/Shared/WebHitTestResultData.serialization.in: Added.
* Source/WebKit/Shared/mac/WebHitTestResultData.mm: Removed.
* Source/WebKit/Shared/mac/WebHitTestResultPlatformData.serialization.in: Added.
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/mac/WKImmediateActionController.mm:
(-[WKImmediateActionController _animationControllerForDataDetectedText]):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
(WebKit::WebPage::performImmediateActionHitTestAtLocation):

Canonical link: <a href="https://commits.webkit.org/257087@main">https://commits.webkit.org/257087@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ed1c5be4a04c0fa316e128216571c5d1b06e3a43

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/97839 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/7076 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/31017 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/107328 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/7513 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/35849 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/90219 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/103964 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/103478 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/5652 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/84466 "Built successfully") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/32600 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [❌ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/101391 "Found 1 webkitpy python2 test failure: webkitcorepy.tests.file_lock_unittest.FileLockTestCase.test_locked_timeout") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/87523 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/89271 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/75515 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/1060 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/20696 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/82/builds/1049 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/22207 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/5878 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/44654 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2418 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/2312 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/41609 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/Shared/WebHitTestResultData.cpp:
```diff
@@ -109,124 +109,44 @@ WebHitTestResultData::WebHitTestResultData(const HitTestResult& hitTestResult, b
     }
 }
+#if PLATFORM(MAC)
+WebHitTestResultData::WebHitTestResultData(const String& absoluteImageURL, const String& absolutePDFURL, const String& absoluteLinkURL, const String& absoluteMediaURL, const String& linkLabel, const String& linkTitle, const String& linkSuggestedFilename, bool isContentEditable, const WebCore::IntRect& elementBoundingBox, const WebKit::WebHitTestResultData::IsScrollbar& isScrollbar, bool isSelecte ...
```
This could just have PLATFORM(MAC) around the one parameter like in the header.  Less redundancy.

gavin-apple committed:
Port WebHitTestResultData to the new IPC serialization format
https://bugs.webkit.org/show_bug.cgi?id=248079
rdar://102518135

Reviewed by Alex Christensen.

Port WebHitTestResultData to the new IPC serialization format and add
support for SoftLinkedClasses in the GeneratedSerializers.

* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Scripts/generate-serializers.py:
(sanitize_string_for_variable_name):
(decode_type):
(parse_serialized_types):
* Source/WebKit/Shared/WebHitTestResultData.cpp:
(WebKit::WebHitTestResultPlatformData::WebHitTestResultPlatformData):
(WebKit::WebHitTestResultData::WebHitTestResultData):
(WebKit::WebHitTestResultData::~WebHitTestResultData):
(WebKit::WebHitTestResultData::getImageSharedMemoryHandle const):
(WebKit::WebHitTestResultData::encode const): Deleted.
(WebKit::WebHitTestResultData::decode): Deleted.
(WebKit::WebHitTestResultData::platformEncode const): Deleted.
(WebKit::WebHitTestResultData::platformDecode): Deleted.
* Source/WebKit/Shared/WebHitTestResultData.h:
* Source/WebKit/Shared/WebHitTestResultData.serialization.in: Added.
* Source/WebKit/Shared/mac/WebHitTestResultData.mm: Removed.
* Source/WebKit/Shared/mac/WebHitTestResultPlatformData.serialization.in: Added.
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/mac/WKImmediateActionController.mm:
(-[WKImmediateActionController _animationControllerForDataDetectedText]):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
(WebKit::WebPage::performImmediateActionHitTestAtLocation):

Canonical link: https://commits.webkit.org/257087@main
